lti_consumer/models.py

```diff
@@ -778,7 +777,6 @@ class CourseAllowPIISharingInLTIFlag(ConfigurationModel):
     course_id = CourseKeyField(max_length=255, db_index=True)
 
     @classmethod
-    @request_cached
```

Comment:
The caching was not working correctly. It appeared that the cached value was not being recomputed on a per-request basis, which meant that the cached value was not successfully being updated. Also, the initial cached value was computed incorrectly. The effect was that the PII sharing XBlock fields were always being displayed in Studio and ``CourseAllowPIISharingInLTIFlag`` instances were not being created for courses that had PII sharing XBlocks that did not already have ``CourseAllowPIISharingInLTIFlag`` instances.

This fixes the backwards compatibility of the ``CourseAllowPIISharingInLTIFlag``. This means that ``CourseAllowPIISharingInLTIFlag`` will now be created for all courses that contain an LTI XBlock that shares username and/or email for which a ``CourseAllowPIISharingInLTIFlag`` does not already exist.

Time was spent on determining why the caching was failing, but it was challenging. If the issue is in the ``request_cached`` decorator, then the impact of changing it is large - it's used elsewhere in the platform. The need for a fix was urgent, and the performance enhancement of leaving the caching in doesn't justify delaying a fix.

Reply:
This is only called by studio anyway right? So the performance impact here doesn't seem all that concerning.

Reply:
Right now, yes. Once we add the LTI configuration service to the right places, it will be called from the LMS as well.